scipy/stats/_stats_py.py

```diff
@@ -4780,8 +4794,11 @@ def statistic(x, y, axis):
     pvalue = _get_pvalue(r, dist, alternative, xp=xp)
 
     mask = (n == 2)   #  return exactly 1.0 or -1.0 values for n == 2 case as promised
-    def special_case(r): return xp.where(xp.isnan(r), xp.nan, xp.ones_like(r))
-    r = xpx.apply_where(mask, (r,), xp.round, fill_value=r)
+    # data-apis/array-api-extra#196
+    mxp = array_namespace(r._meta) if is_dask(xp) else xp    
+    def special_case(r):
+        return mxp.where(mxp.isnan(r), mxp.nan, mxp.ones_like(r))
+    r = xpx.apply_where(mask, (r,), mxp.round, fill_value=r)
```

Comment:
IIRC, this should work now, too, right?
```suggestion
    r = xpx.apply_where(mask, r, mxp.round, fill_value=r)
```